services/typescript-service.ts

```diff
@@ -0,0 +1,316 @@
+				// Create typeScript command file
+				let typeScriptCommandsFilePath = path.join(temp.mkdirSync("typeScript-compilation"), "tscommand.txt");
+				let typeScriptCompilerOptions = this.getTypeScriptCompilerOptionsAsArguments(compilerOptions);
+				let typeScriptDefinitionsFiles: string[];
+
+				if (!this.hasTsConfigFile(projectDir).wait()) {
+					typeScriptDefinitionsFiles = this.getDefaultTypeScriptDefinitionsFiles(options.pathToDefaultDefinitionFiles).wait();
+				}
+
+				this.$fs.writeFile(typeScriptCommandsFilePath, this.typeScriptFiles.concat(typeScriptDefinitionsFiles).concat(typeScriptCompilerOptions).join(" ")).wait();
+
+				let typeScriptCompilerSettings = this.getTypeScriptCompiler().wait();
+
+				// Log some messages
+				this.$logger.out("Compiling...".yellow);
+				_.each(this.typeScriptFiles, file => {
+					this.$logger.out(`### Compile ${file}`.cyan);
+				});
+				this.$logger.out(`Using tsc version ${typeScriptCompilerSettings.version}`.cyan);
+
+				// Core compilation
+				return this.runTranspilation(projectDir, typeScriptCompilerSettings.pathToCompiler, { typeScriptCommandsFilePath }).wait();
+			} else {
+				return this.transpileWithDefaultOptions(projectDir, compilerOptions).wait();
+			}
+		}).future<string>()();
+	}
+
+	// Uses tsconfig.json if it exists
+	public transpileWithDefaultOptions(projectDir: string, options?: ITypeScriptTranspileOptions): IFuture<string> {
+		return ((): string => {
+			options = options || {};
+			let compilerOptions = this.getCompilerOptions(projectDir, options).wait();
+			this.noEmitOnError = compilerOptions.noEmitOnError;
+			let typeScriptCompilerSettings = this.getTypeScriptCompiler().wait();
+			this.$logger.out(`Using tsc version ${typeScriptCompilerSettings.version}`.cyan);
+
+			// Core compilation
+			return this.runTranspilation(projectDir, typeScriptCompilerSettings.pathToCompiler, { compilerOptions }).wait();
+		}).future<string>()();
+	}
+
+	public getTypeScriptFiles(projectDir: string): IFuture<ITypeScriptFiles> {
+		return ((): ITypeScriptFiles => {
+			// Skip root's node_modules
+			let rootNodeModules = path.join(projectDir, "node_modules");
+			let projectFiles = this.$fs.enumerateFilesInDirectorySync(projectDir,
+				(fileName: string, fstat: IFsStats) => fileName !== rootNodeModules);
+			let typeScriptFiles = _.filter(projectFiles, file => path.extname(file) === ".ts");
+			let definitionFiles = _.filter(typeScriptFiles, file => _.endsWith(file, ".d.ts"));
+			return { definitionFiles: definitionFiles, typeScriptFiles: typeScriptFiles };
+		}).future<ITypeScriptFiles>()();
+	}
+
+	public isTypeScriptProject(projectDir: string): IFuture<boolean> {
+		return ((): boolean => {
+			let typeScriptFiles = this.getTypeScriptFiles(projectDir).wait();
+
+			if (typeScriptFiles.typeScriptFiles.length > typeScriptFiles.definitionFiles.length) { // We need this check because some of non-typescript templates(for example KendoUI.Strip) contain typescript definition files
```

Comment:
Shouldn't this check simply be `if (typeScriptFiles.typeScriptFiles.length > 0)`?
With the current state if I have a single TypeScript file and a single TypeScript definition file `isTypeScriptProject` would return `false`